tstest/integration/vms/vms_test.go

```diff
@@ -415,20 +446,39 @@ func TestVMIntegrationEndToEnd(t *testing.T) {
 		for port := range ipMap {
 			port := port
 			t.Run(port, func(t *testing.T) {
-				config := &ssh.ClientConfig{
-					User:            "ts",
-					Auth:            []ssh.AuthMethod{ssh.PublicKeys(signer), ssh.Password("hunter2")},
-					HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+				t.Parallel()
+				time.Sleep(5 * time.Second)
+
+				hostport := fmt.Sprintf("127.0.0.1:%s", port)
+
+				const maxRetries = 5
+				var working bool
+				for i := 0; i < maxRetries; i++ {
+					conn, err := net.Dial("tcp", hostport)
+					if err == nil {
+						working = true
+						conn.Close()
+						break
+					}
+
+					time.Sleep(5 * time.Second)
 				}
 
-				cli, err := ssh.Dial("tcp", fmt.Sprintf("127.0.0.1:%s", port), config)
-				if err != nil {
-					t.Fatalf("can't dial 127.0.0.1:%s: %v", port, err)
+				if !working {
+					t.Fatalf("can't connect to %s, tried %d times", hostport, maxRetries)
 				}
-				defer cli.Close()
 
-				t.Parallel()
 				t.Logf("about to ssh into 127.0.0.1:%s", port)
+				cli, err := ssh.Dial("tcp", hostport, &ssh.ClientConfig{
+					User:            "root",
+					Auth:            []ssh.AuthMethod{ssh.PublicKeys(signer), ssh.Password("hunter2")},
```

Comment:
move hunter2 to a const somewhere and reuse it from other places

Reply:
Done